### ISSUE TITLE
Synthesizer.WordBoundary method not working to get word audio duration (including latest SDK)

### ISSUE THREAD

**Issue by coder-kl** [Azure-Samples/cognitive-services-speech-sdk#1843]
**Describe the bug**
synthesizer.WordBoundary event is not firing for Azure function or Console app.

**To Reproduce**
The sample code is provided at end of the post. I posted this issue last year on learn.microsoft.com forum and team could not reproduced. They suggested to contact SDK team with a log file. I have attached the log file with the post too
https://learn.microsoft.com/en-us/answers/questions/849161/azure-text-to-speech-synthesizer-wordboundary-meth
https://learn.microsoft.com/en-us/answers/questions/1181912/synthesizer-wordboundary-method-still-not-working?comment=question#newest-question-comment

**Expected behavior**
I would expect to capture word boundary events, so I can implement text highlight as part of my read aloud module in my app. 

**Version of the Cognitive Services Speech SDK**
[1.25.0](https://www.nuget.org/packages/Microsoft.CognitiveServices.Speech/1.25.0)
I have tried the same code with previous version of SDKs, however it did not work either. 

**Platform, Operating System, and Programming Language**
 - OS: Windows 10
 - Hardware - x64
 - Programming language: C#
 - Browser: NA

**Additional context**
 - Error messages, stack trace, sample code to reproduce the issue - NA, no error
 - Please provide full Speech SDK Log - is attached.
 - Sample Code:
 
private static async Task SynthesisWordBoundaryEventAsync(string text, SpeechConfig config) 
{
        using (var synthesizer = new SpeechSynthesizer(config, null as AudioConfig))  
        {  
            // Subscribes to word boundary event  
            synthesizer.WordBoundary += (s, e) =>  
            {  
                // The unit of e.AudioOffset is tick (1 tick = 100 nanoseconds), divide by 10,000 to convert to milliseconds.  
                Console.WriteLine($"Word boundary event received. Audio offset: " +  
                        $"{(e.AudioOffset + 5000) / 10000}ms, text offset: {e.TextOffset}, word length: {e.WordLength}.");  
            };  

            using (var result = await synthesizer.SpeakTextAsync(text))  
            {  
                if (result.Reason == ResultReason.SynthesizingAudioCompleted)  
                {  
                    Console.WriteLine($"Speech synthesized for text .");  
                    var audioData = result.AudioData;  
                    Console.WriteLine($"{audioData.Length} bytes of audio data received for text [{text}]");  
                }  
                else if (result.Reason == ResultReason.Canceled)  
                {  
                    var cancellation = SpeechSynthesisCancellationDetails.FromResult(result);  
                    Console.WriteLine($"CANCELED: Reason={cancellation.Reason}");  

                    if (cancellation.Reason == CancellationReason.Error)  
                    {  
                        Console.WriteLine($"CANCELED: ErrorCode={cancellation.ErrorCode}");  
                        Console.WriteLine($"CANCELED: ErrorDetails=[{cancellation.ErrorDetails}]");  
                        Console.WriteLine($"CANCELED: Did you update the subscription info?");  
                    }  
                }  
            }  
        }  
    }  

Let me know if you need any additional details. 

Thanks,
Samir
[TextToSpeechLog.txt](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/10799548/TextToSpeechLog.txt)

**Comment by yulin-li** [comment]
Hi, could you share the codes when you initialize your `SpeechConfig`? I noticed that you are explicitly using REST api, while word boundary events are only available via websocket.

**Comment by yulin-li** [comment]
Could you try `var config = SpeechConfig.FromSubscription("YourSubscriptionKey", "eastus");` ?

**Comment by coder-kl** [comment]
Thank you Yulin  for your feedback. It worked :)

I was using SpeechConfig.FromSubscriptionUrl method. After switching to SpeechConfig.FromSubscription generated the the events and I was able to capture duration for each words. Thank you so much. 

Secondly, I am planning to use this method in Azure Function. I have tested with console app today. Do I need to consider anything special for Azure function? Or the same code should work.

Final question: Will VisemeReceived work with the same configuration? Or any special settings I need to do to capture VisemeReceived events. I haven't tried yet, however I am planning to look into later in the year. 

Samir
